### PR TITLE
Expose regions in the non-standard partitions

### PR DESCRIPTION
**Pull request by rli**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
* don't think we have a good strategy for cross-partition testing with real credentials

## Related Issue(s)
#1405 #699

## Screenshots (if appropriate)
out

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kiiadi** [comment]
I'm not sure I like having the partitions thing exposed so publicly - for *most* customers they don't care. I'd rather have a 'Switch Partition' option deeper in the menu structure...

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultProjectAccountSettingsManager.kt [hunk 1]
```diff
@@ -25,6 +26,7 @@ data class ConnectionSettingsState(
 class DefaultProjectAccountSettingsManager(private val project: Project) : ProjectAccountSettingsManager(project),
     PersistentStateComponent<ConnectionSettingsState> {
     override fun getState(): ConnectionSettingsState = ConnectionSettingsState(
+        activePartition = selectedPartition?.id,
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultProjectAccountSettingsManager.kt]: Do we need this? Or can we infer it from active region?

**Comment by rli** [comment]
Mocked up some alternatives in #1625

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultProjectAccountSettingsManager.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultProjectAccountSettingsManager.kt]: Probably cleaner to infer, I don't think it's worth persisting any transient state resulting from the user switching between partitions.

**Pull request merged**